src/storage.rs

```diff
@@ -528,47 +579,41 @@ mod test {
         let storage = MemStorage::new();
         storage.wl().entries = ents;
 
-        let wresult = Ok(4);
-        let result = storage.first_index();
-        if result != wresult {
-            panic!("want {:?}, got {:?}", wresult, result);
-        }
-
-        storage.wl().compact(4).expect("compact failed");
-        let wresult = Ok(5);
-        let result = storage.first_index();
-        if result != wresult {
-            panic!("want {:?}, got {:?}", wresult, result);
-        }
+        assert_eq!(storage.first_index(), Ok(3));
+        storage.wl().compact(4).unwrap();
+        assert_eq!(storage.first_index(), Ok(4));
     }
 
     #[test]
     fn test_storage_compact() {
         setup_for_test();
         let ents = vec![new_entry(3, 3), new_entry(4, 4), new_entry(5, 5)];
         let mut tests = vec![
-            (2, Err(RaftError::Store(StorageError::Compacted)), 3, 3, 3),
```

Comment:
So this no longer returns an error? Why?

Reply:
I think in this case an error is unnecessary. Please consider this case:
```rust
storage.compact(100);
storage.compact(80);
```
Firstly we compact to 100, and then compact to 80. Previously it will return an Error, but I think we can just do nothing internally, and return `Ok`. What do you think?

Reply:
I think just no-oping it is fine. :)